apiserver/client/client.go

```diff
@@ -392,11 +395,27 @@ func (c *Client) ProvisioningScript(args params.ProvisioningScriptParams) (param
 		icfg.EnableOSRefreshUpdate = cfg.EnableOSRefreshUpdate()
 	}
 
-	result.Script, err = sshprovisioner.ProvisioningScript(icfg)
+	osSeries, err := series.GetOSFromSeries(icfg.Series)
 	if err != nil {
-		return result, common.ServerError(errors.Annotate(
-			err, "getting provisioning script",
-		))
+		return result, common.ServerError(errors.Annotatef(err,
+			"cannot decide which provisioning script to generate based on this series %q", icfg.Series))
+	}
+
+	switch osSeries {
```

Comment:
this could be simplified to:

```go
getProvisioningScript := sshprovisioner.ProvisioningScript
if osSeries == os.Windows {
    getProvisioningScript = winrmprovisioner.ProvisioningScript
}
result.Script, err = getProvisioningScript(icfg)
if err != nil {
    return result, common.ServerError(errors.Annotate(
        err, "getting provisioning script",
    ))
}
```